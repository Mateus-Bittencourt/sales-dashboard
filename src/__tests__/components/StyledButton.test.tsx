import { StyledButton } from '@components'
import { render } from '@testing-library/react'
import type { Theme } from '@types'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@resources/themesList'

describe('StyledButton Component', () => {
  const renderComponent = (theme: Theme, className?: string, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledButton className={className} {...props} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`when using ${name}`, () => {
      it('should match snapshot with alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match snapshot with primary class', () => {
        const { asFragment } = renderComponent(theme, 'primary')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match snapshot with borderless-alert class', () => {
        const { asFragment } = renderComponent(theme, 'borderless-alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match snapshot with disabled prop', () => {
        const { asFragment } = renderComponent(theme, 'primary', {
          disabled: true,
        })
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
