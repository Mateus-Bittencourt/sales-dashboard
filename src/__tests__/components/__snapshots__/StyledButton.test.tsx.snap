// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component when using Dark Theme should match snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Dark Theme should match snapshot with borderless-alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Dark Theme should match snapshot with disabled prop 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Dark Theme should match snapshot with primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #313649;
  color: #6d7b8e;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #313649;
  color: #6d7b8e;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Light Theme should match snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Light Theme should match snapshot with borderless-alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Light Theme should match snapshot with disabled prop 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Component when using Light Theme should match snapshot with primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s ease,color 0.3s ease;
  width: 100%;
}

.c0.primary {
  background-color: #0c70f2;
  color: #fff;
}

.c0.primary:hover {
  background-color: #0061de;
}

.c0.alert {
  background-color: #e80000;
  color: #fff;
}

.c0.alert:hover {
  background-color: #d80000;
}

.c0.borderless-alert {
  background-color: none;
  color: #e80000;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #d80000;
}

.c0:disabled {
  background-color: #ccc;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  background-color: #ccc;
  color: #666;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;
